### ISSUE TITLE
BUG: duplicate shifted holiday in California

### ISSUE THREAD

**Issue by joooeey** [peopledoc/workalendar#528]
``` python
from workalendar.usa import California

cal = California()

cal.holidays(2019)
Out[13]: 
[(datetime.date(2019, 1, 1), 'New year'),
 (datetime.date(2019, 1, 21), 'Birthday of Martin Luther King, Jr.'),
 (datetime.date(2019, 2, 18), "Washington's Birthday"),
 (datetime.date(2019, 3, 31), 'Cesar Chavez Day'),
 (datetime.date(2019, 4, 1), 'Cesar Chavez Day (Observed)'),  # <===============
 (datetime.date(2019, 4, 1), 'Cesar Chavez Day (Observed)'),  # <===============
 (datetime.date(2019, 5, 27), 'Memorial Day'),
 (datetime.date(2019, 7, 4), 'Independence Day'),
 (datetime.date(2019, 9, 2), 'Labor Day'),
 (datetime.date(2019, 11, 11), 'Veterans Day'),
 (datetime.date(2019, 11, 28), 'Thanksgiving Day'),
 (datetime.date(2019, 11, 29), 'Thanksgiving Friday'),
 (datetime.date(2019, 12, 25), 'Christmas Day')]
```

**Note that Cesar Chavez Day (Observed) is duplicated.**

This does not occur for 2017, 2018, 2020, and 2021. I've just tested those five years. The same issue occurs with 2019 in the CaliforniaEducation calendar.

I noticed this issue because it breaks a version of my test for PR #527.

**Comment by brunobord** [comment]
thank you for the bug report, very "interesting" bug. I'll get into it as soon as I can.

**Comment by joooeey** [comment]
I just looked into this. The shift day is computed twice (and differently) in `workalendar/usa/core.py`. Saturday is shifted to Friday and Sunday to Monday in `def shift` and if the holiday falls on a Sunday it is shifted to Monday in `def get_cesar_chavez_days`.

[Here's some info on how state holidays in California work.](https://www.calhr.ca.gov/employees/Pages/state-holidays.aspx) It says about the shift holidays:

>  *When a holiday falls on a Saturday, employees shall receive holiday credit.
> **When a holiday falls on a Sunday, the holiday is observed on the following Monday

I could not find any source stating that Cesar Chavez Day is shifted any differently than other holidays.

Hence, I think this bug should be solved by not doing any special shifting for Cesar Chavez Day, i.e. have a `def get_cesar_chavez_day` function just like all the other holiday functions that returns a holiday on March 31 and no extra shift day.

However I noticed two other, separate issues:

- As per the link above, California state holidays falling on a Saturday are not actually shifted to Friday but result in holiday credit instead. In `usa/core.py` we shift them to Friday.
- Per Wikipedia Cesar Chavez Day is a state holiday in a few states, in `workalendar` we only have it in California. So we're inconsistent. Also, it doesn't seem to be a holiday for most workers in California.

I'm not gonna touch these two bullet points as this looks like a job for an American.

**Comment by brunobord** [comment]
Bug fixed, you can use the latest (10.4.0) release:

* PyPI: https://pypi.org/project/workalendar/10.4.0/
* Release notes: https://github.com/peopledoc/workalendar/releases/tag/10.4.0

I'm still working my way on the iCal export feature, soon to be ready.

**Comment by joooeey** [comment]
> Effectively, Chavez Day seems to be an exception when it comes to weekend-shifting.

It would be sweet to document the source for this statement. To me it looked like this was an implementation mistake. I couldn't find any official info that shifting works differently for Cesar Chavez Day.
